scripts/compactor_runner.py

```diff
@@ -282,17 +244,71 @@ def _run_corfu_compactor(self):
+
+        server_address = compactor_config["ServerAddress"]
+        if not is_saas_hostname_present and not self._config.hostname:
+            if self._config.network_interface and self._config.network_interface_version:
+                self._print_and_log("WARNING: _run_corfu_compactor: Couldn't load the hostname from args or config "
+                                    "file. Please check the parameters. Continuing self IP discovery with {}"
+                                    " interface and {} version.".format(self._config.network_interface,
+                                                                        self._config.network_interface_version))
+                self._config.hostname = self._resolve_ip_address(
+                    network_interface=self._config.network_interface,
+                    network_interface_version=self._config.network_interface_version
+                )
+            elif server_address:
+                self._print_and_log("_run_corfu_compactor: Found the server_address from config file: {}"
+                                    .format(server_address))
+                if server_address["Hostname"]:
+                    self._config.hostname = server_address["Hostname"]
+                # Do self ip discovery since hostname is not configured
+                elif server_address["NetworkInterfaceVersion"] \
+                        and server_address["NetworkInterface"]:
+                    self._print_and_log("WARNING: _run_corfu_compactor: Couldn't load the hostname from args or config "
+                                        "file. Please check the parameters. Continuing self IP discovery with {}"
+                                        " interface and {} version.".format(server_address["NetworkInterface"],
+                                                                            server_address["NetworkInterfaceVersion"]))
+                    self._config.hostname = self._resolve_ip_address(
+                        network_interface=server_address["NetworkInterface"],
+                        network_interface_version=server_address["NetworkInterfaceVersion"]
+                    )
+                else:
+                    raise RuntimeError("_run_corfu_compactor: Couldn't load the hostname or network interface info"
+                                       " from args or config file. Please check the configurations.")
+            else:
+                # When server address param is not there in the config file or hostname is not present in the args
+                raise RuntimeError("_run_corfu_compactor: Couldn't load the server_address from args or config file."
+                                   " Please check the configurations.")
+
+        if not self._config.corfu_port and server_address and server_address["Port"]:
+            self._config.corfu_port = str(server_address["Port"])
+        else:
```

Comment:
Even if self._config.corfu_port is passed as parameter, this else will be hit and it raises the exception